packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1596,4 +1795,25 @@ class _SelectableFragment with Selectable, ChangeNotifier {
       );
     }
   }
+
+  @override
+  TextSelection getLineAtOffset(TextPosition position) {
+    final TextRange line = paragraph._getLineAtOffset(position);
+    final int start = line.start.clamp(range.start, range.end); // ignore_clamp_double_lint
+    final int end = line.end.clamp(range.start, range.end); // ignore_clamp_double_lint
```

Comment:
Why can't you use clampDouble for these?

Reply:
This is in type, so we should not use clampDouble.
 
the purpose of this ignore_clamp_double_lint is to make sure we DON'T use the .clamp method on `double`. The problem is that our current lint can't distinguish between .clamp called on `int` or `double` or any `num` subclass. So the current suggestion is to add this ignore if you are sure the clamp is called on non-double data type.